zk_toolbox/crates/zk_inception/src/utils/ports.rs

```diff
@@ -0,0 +1,364 @@
+use std::{
+    collections::{HashMap, HashSet},
+    fmt,
+    ops::Range,
+    path::Path,
+};
+
+use anyhow::{Context, Result};
+use config::{EcosystemConfig, PortsConfig};
+use serde_yaml::Value;
+use xshell::Shell;
+
+use crate::defaults::{
+    LOCAL_HTTP_RPC_PORT, LOCAL_WS_RPC_PORT, OBSERVABILITY_PORT, PORT_RANGE, POSTGRES_DB_PORT,
+};
+
+#[derive(Default)]
+pub struct EcosystemPorts {
+    pub ports: HashMap<u16, Vec<String>>,
+}
+
+impl EcosystemPorts {
+    /// Creates a new EcosystemPorts instance with default port configurations.
+    /// These ports are not configured in ecosystem yaml file configs, but are
+    /// commonly used or pre-assigned within the ecosystem.
+    pub fn with_default_ports() -> Self {
+        let mut ports = HashMap::new();
+        ports.insert(OBSERVABILITY_PORT, vec!["Observability".to_string()]);
+        ports.insert(POSTGRES_DB_PORT, vec!["PostgreSQL".to_string()]);
+        ports.insert(LOCAL_HTTP_RPC_PORT, vec!["Local HTTP RPC".to_string()]);
+        ports.insert(LOCAL_WS_RPC_PORT, vec!["Local WebSockets RPC".to_string()]);
```

Comment:
do we need it if we scan docker compose?

Reply:
No, I don't think we need it anymore

Reply:
We may or may not have era-observability folder, so I'd leave OBSERVABILITY_PORT (3000).